libsolidity/interface/StandardCompiler.cpp

```diff
@@ -618,6 +655,8 @@ std::variant<OptimiserSettings, Json::Value> parseOptimizerSettings(Json::Value
 				return *error;
 			if (auto error = checkOptimizerDetailSteps(details["yulDetails"], "optimizerSteps", settings.yulOptimiserSteps))
 				return *error;
+			if (auto error = checkOptimizerDetailCleanupSteps(details["yulDetails"], "optimizerSteps", settings.yulOptimiserCleanupSteps))
+				return *error;
```

Comment:
```suggestion
			if (auto error = checkOptimizerDetailSteps(details["yulDetails"], "optimizerSteps", settings.yulOptimiserSteps, settings.yulOptimiserCleanupSteps))
				return *error;
```

It'd probably be more straightforward and entail less duplication to handle both of them together, wouldn't it?

Reply:
Yup. Wasn't sure which approach to take; on one hand it would certainly be less redundant to simply extend the function prototype and accept two output strings (for settings), but on the other hand, all of these 'helper' functions handle one field, and so the current implementation 'fits' better with the rest of the codebase. I'm fine with either, so I'll either change it or leave it as is depending on the consensus.

Reply:
You can turn that around: all the other functions handle one json field entirely, while this, if split into two functions, handles only half of the json field :-). So it's *some* non-uniformity either way, hence I'd argue the less redundant one is better :-).